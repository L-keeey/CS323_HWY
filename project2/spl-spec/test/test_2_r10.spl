struct Person
{
  int friends[10][13][14];
  int name;
};
int test_2_r10()
{
  struct Person tom;
  struct Person people[10];
  int i = 0;
  while (i < 10)
  {
    people[i].name = i;
    tom.friends[i] = i;
    i = i + 1;
  }
  return tom.name[i-1];
}
