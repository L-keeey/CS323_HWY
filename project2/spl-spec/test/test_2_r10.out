in def var
in def var
def struct
start field list.
generate field list.
non-empty structure
get field list from node
0x56407e8e7ce0
0x56407e8e59e0
0x56407e8e58b0
0x56407e8e5650
0x56407e8e5190
0x56407e8e4cd0
0x56407e8e4810
success get name from declist
friends
prim/array type list node
Generate a array
Dec
initial
4
14
rec gen arr
4
13
rec gen arr
4
10
rec gen arr
complete gen arr
0x56407e8e83b0
get field list from node
0x56407e8e7bb0
0x56407e8e7660
0x56407e8e7530
0x56407e8e72d0
success get name from declist
name
prim/array type list node
Generate a prim type
add a field list
field list size
2
0x56407e8e82d0
def complete, then cal hash
cal field list hash
cal array hash
0x56407e8e83b0
foo1
foo1
foo1
******
cal field list hash
816743588
struct def complete
in def var
in def var
Error type 3 at Line 9: variable is redefined.
in def var
checking value
0x56407e8eb2b0
check is same type
0
0
foo
0
0
1
Error type 1 at Line 13: variable is used without a definition.
